minerva/tools/fetch_html.py

```diff
@@ -34,22 +36,38 @@ async def fetch_html(url: str) -> str:
   Use this tool when you need to visit a website and fetch its content.
   """
 
-  async with httpx.AsyncClient() as client:
-    response = await client.get(
-      url,
-      timeout=TIMEOUT_SEC,
-      headers={
-        "User-Agent": "Minerva AI Bot - (https://github.com/move-fast-and-break-things/minerva)"
-      },
-      follow_redirects=True,
-    )
-    response.raise_for_status()
-
-    # raise if the response is not text
-    if not response.headers.get("content-type", "").startswith("text/"):
-      raise ValueError(f"Unexpected content type: {response.headers.get('content-type')}")
-
-    return clean_html(response.text)
+  # async with httpx.AsyncClient() as client:
+  #   response = await client.get(
+  #     url,
+  #     timeout=TIMEOUT_SEC,
+  #     headers={
+  #       "User-Agent": "Minerva AI Bot - (https://github.com/move-fast-and-break-things/minerva)"
+  #     },
+  #     follow_redirects=True,
+  #   )
+  #   response.raise_for_status()
+
+  #   # raise if the response is not text
+  #   if not response.headers.get("content-type", "").startswith("text/"):
+  #     raise ValueError(f"Unexpected content type: {response.headers.get('content-type')}")
+
+  #   return clean_html(response.text)
+
+  async with async_playwright() as p:
+        browser = await p.chromium.launch(headless=True)
+        page = await browser.new_page()
+        page.set_default_timeout(TIMEOUT_SEC * 1000)
+
+        # Set a user agent like before
+        await page.set_extra_http_headers({
+            "User-Agent": "Minerva AI Bot - (https://github.com/move-fast-and-break-things/minerva)"
+        })
+
+        await page.goto(url, wait_until="networkidle")  # wait until JS is mostly done
```

Comment:
Waiting for `networkidle` may never resolve because some websites do continuous polling in the background.

[Docs:](https://playwright.dev/docs/api/class-page#page-wait-for-load-state)

<img width="1026" height="336" alt="image" src="https://github.com/user-attachments/assets/87cee34f-5003-47e7-ac9b-dec56bee119d" />